components/Calendar.vue

```diff
@@ -0,0 +1,178 @@
+<template>
+    <div class="my-8 container m-auto text-xl">
+        <div class="flex justify-between items-center rounded-lg p-4">
+            <div class="text-2xl pl-2">
+                <button @click="showPreviousMonth" class="mr-6">&#10094;</button>
+                <button @click="showNextMonth">&#10095;</button>
+            </div>
+            <h2 class="text-center">{{ getMonthName(currentMonth) }} {{ currentYear }}</h2>
+            <div class="pr-2">
+                <select v-model="currentYear" @change="createCalendar()" class="bg-transparent">
+                    <option v-for="year in yearRange" :key="year" :value="year">{{ year }}</option>
+                </select>
+            </div>
+      </div>
+      <table class="w-full mt-6">
+        <tr>
+          <th v-for="weekday in weekdays" :key="weekday" class="border-b-2 text-center cursor-pointer py-1 px-1">{{ weekday }}</th>
+        </tr>
+        <tr v-for="week in weeks" :key="week.id">
+          <td v-for="day in week.days" :key="day.date"
+           :class="[
+              isToday(day) ? 'bg-blue-400' : '',
+              day.selected ? 'bg-green-400' : 'bg-white'
+          ]"
+          class="border-2 text-center cursor-pointer py-1 px-1 h-32 w-24 hover:bg-orange-300 hover:shadow-outline"
+          @click="selectDay(day)">
+          {{ day.date }}
+          <div class="text-sm">
+            <ul>
+              <li v-for="(todo, idx) in getEventByDate(day)" :key="todo.id">
+                <p v-if="idx<=maxContainIndex" class="p-2 m-2 bg-blue-100 rounded-lg">
+                    <span :title="todo.name">{{todo.name}}</span>
+                </p>
+              </li>
+            </ul>
+          </div>
+          <div class="mb-2">
+            <span v-if="getEventByDate(day).length>maxContainIndex" class="p-2 mb-2 text-xs bg-orange-200  rounded-lg">Click for More</span>
+          </div>
+      </td>
+        </tr>
+      </table>
+      <InfoModal v-if="showInfoModal" :date="showDate.date" :showDateTitle="showDate.title" :showInfoModal="showInfoModal" @close="showInfoModal=false"/>
+    </div>
+  </template>
+  
+<script setup lang="ts">
+const maxContainIndex = 2;
+const todosStore = useTodosStore();
+const todos = ref(todosStore.todos);
+const showInfoModal = ref<boolean>(false);
+const currentYear = ref<number>(new Date().getFullYear());
+const currentMonth = ref<number>(new Date().getMonth());
+const selectedDay = ref<any>();
+const weekdays = ref<string[]>([
```

Comment:
You can omit the type definition for generics in such cases, ref will automatically infer the type from variable provided